aten/src/ATen/native/cuda/SegmentReduce.cu

```diff
@@ -23,6 +23,19 @@ struct CustomMax {
   }
 };
 
+struct CustomSum {
```

Comment:
do you need special nan handling here? `a+b` should propagate nans.

Reply:
Given the last time issues (this part and also making it work for rocm) with this for max, just followed safe path that I know works.

Reply:
`max` is a known bug on rocm, sum is a standard operation